### PR TITLE
Add home component

### PR DESCRIPTION
**Pull request by jmhooper**
This is a WIP commit to change the content displayed in `index.html` into a component. This will pave the way for us to address issues like #801.

At moment, my thinking is to move the static HTML into a string in the component, and render that when the component mounts. When the component receives props, we check for a user. If a user is present, we redirect to `/sites`.

### CODE REVIEW

**Comment by jmhooper** [comment]
I've updated the tests and the commit message so this should be ready to go. The commit message:

```
This commit adds a new component which is meant to be displayed at the root before the user logs in. As soon as a user is present in the state, the component redirects to `/site`.

The component has a bunch of static HTML, much of which does not elegantly translate to JSX, so the HTML is stored alongside the component in `home.html`. The HTML is loaded as a string using `stringify` and then set when the component is rendered.
```

File: frontend/reducers/user.js [hunk 1]
```diff
@@ -5,6 +5,10 @@ import {
 export default function user(state = false, action) {
   switch (action.type) {
   case USER_RECEIVED:
+    if (!action.user) {
+      return false
```
Comment (brandocalrissian) [inline frontend/reducers/user.js]: TIL `false.foo` is something that can actually have a useful meaning. We're getting a little lucky here, but it works.

File: frontend/components/home.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  redirectForLoggedInUser() {
+    if (this.props.storeState.user) {
+      pushHistory("/sites")
+    }
+  }
+
+  render() {
+    return <div dangerouslySetInnerHTML={{ __html: homeHTML }}/>
```
Comment (jmhooper) [inline frontend/components/home.js]: Yeah, there's a few things that trip up react. Also note the SVG markup in there.

**Comment by brandocalrissian** [comment]
Will merge when the CI comes back green.

**Pull request merged**